packages/tldraw/src/state/shapes/ImageUtil/ImageUtil.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import * as React from 'react'
+import { Utils, HTMLContainer } from '@tldraw/core'
+import { ImageShape, TDShapeType, TDMeta } from '~types'
+import { TDShapeUtil } from '../TDShapeUtil'
+import { styled } from '~styles'
+import {
+  defaultStyle,
+  getShapeStyle,
+  getBoundsRectangle,
+  transformRectangle,
+  transformSingleRectangle,
+} from '~state/shapes/shared'
+
+type T = ImageShape
+type E = HTMLDivElement
+
+export class ImageUtil extends TDShapeUtil<T, E> {
+  type = TDShapeType.Image as const
+
+  isAspectRatioLocked = true
+
+  canBind = true
+
+  getShape = (props: Partial<T>): T => {
+    return Utils.deepMerge<T>(
+      {
+        id: 'id',
+        type: TDShapeType.Image,
+        name: 'Image',
+        parentId: 'page',
+        childIndex: 1,
+        point: [0, 0],
+        size: [1, 1],
+        rotation: 0,
+        style: defaultStyle,
+        data: '',
+      },
+      props
+    )
+  }
+
+  Component = TDShapeUtil.Component<T, E, TDMeta>(({ shape, events }, ref) => {
+    const rImg = React.useRef<HTMLImageElement>(null)
+
+    const onImageLoad = () => {
+      if (rImg && rImg.current) {
+        const { width, height } = rImg?.current
+        shape.size = [width, height]
```

Comment:
This is NO NO
I will replace with something better way to mutate shape.

Steve's suggestion "so (onShapeChange({ id: shape.id, size: [width, height] }) should work"